### PR TITLE
TribitsExample[Project2|App2]: Add support for optional package and TPL dependencies (#299)

### PR DESCRIPTION
**Pull request by bartlettroscoe**
Updated `TribitsExampleProject2` and `TribitsExampleApp2` for optional dependencies being disabled.  This required adding the generation of configured `<Package>_config.h` files and adding `<Package>_ENABLE_<dep>` vars to the `<Package>Config.cmake` files.  (This shows how we can export info about packages to their `<Package>Config.cmake`  files so that downstream packages can get access to that info.  There is other types of info we will need to do this for as well as we allow building and installing TriBITS packages (like in Trilinos) individually.

And with this PR in place, now I can add support for TPL dependencies!

This part of #299.

### CODE REVIEW

File: test/core/ExamplesUnitTests/TribitsExampleProject2_Tests.cmake [hunk 1]
```diff
@@ -264,6 +286,130 @@ TribitsExampleProject2_find_tpl_parts(SHARED  CMAKE_PREFIX_PATH_ENV)
...
+
+  TribitsExampleProject2_test_setup_header()
+
+  set(tplInstallBaseDir
+    "${TribitsExampleProject2_Tpls_install_${sharedOrStatic}_DIR}")
+
+  set(testNameSuffix "")
+  set(tplLibAndIncDirsArgs "")
+  set(cmakePrefixPathCacheArg "")
+  set(cmakePrefixPathEnvArg "")
```
Comment (rabartlett1972) [inline test/core/ExamplesUnitTests/TribitsExampleProject2_Tests.cmake:299]: This variable is not needed in this function.  Should remove.

**Review by rabartlett1972 — CHANGES_REQUESTED** [review]
I don't see any major issues, just a var to clean up.

File: test/core/ExamplesUnitTests/TribitsExampleProject2_Tests.cmake (hunk 1 above)
Comment (bartlettroscoe) [reply test/core/ExamplesUnitTests/TribitsExampleProject2_Tests.cmake]: All of these unneeded temp vars are removed in new commit https://github.com/TriBITSPub/TriBITS/pull/474/commits/7b552d36100f5b6851cd49493eb3a5617c8c3bb1.

**Review by rabartlett1972 — APPROVED** [review]
Looks good.

**Comment by bartlettroscoe** [comment]
@keitat, you can review some if you want.

**Pull request merged**